libs/librepcb/libraryeditor/newelementwizard/newelementwizardcontext.h

```diff
@@ -129,7 +129,8 @@ class NewElementWizardContext final : public QObject {
   QString                   mElementKeywords;
   QString                   mElementAuthor;
   tl::optional<Version>     mElementVersion;
-  tl::optional<Uuid>        mElementCategoryUuid;
+  tl::optional<Uuid>        mElementRootCategoryUuid;
+  QSet<Uuid>                mElementCategoriesUuid;
```

Comment:
Wouldn't it be simpler to have only the QSet member? I mean, the optional<Uuid> could simply be generated on demand where needed (take one item from the set if it is not empty, else nullopt - it's impossible anyway that the set contains multiple items at the time we need the optional<Uuid>).

And btw, I usually name a list of UUIDs like *ElementUuids* instead of *ElementsUuid* :wink:

```suggestion
  QSet<Uuid>                mElementCategoryUuids;
```